### PR TITLE
Tighten up PyLint directives

### PR DESCRIPTION
**Pull request by cclauss**
## Description ##
Remove PyLint disable directives that the codebase is not violating.  This tightens up the linting of future proposed changes.  Also looked through PyLint issues that were only being flagged once in the code base to see if small changes code make the code compliant.  __Fixed__ via code changes:
```
************* Module setup
python/setup.py:32:0: C0413: Import "from setuptools import find_packages" should be placed at the top of the module (wrong-import-position)
************* Module mxnet.executor_manager
python/mxnet/executor_manager.py:130:24: C0121: Comparison to False should be 'expr' or 'expr is not False' (singleton-comparison)
************* Module mxnet.model
python/mxnet/model.py:136:16: W1508: os.getenv default type is builtins.int. Expected str or None. (invalid-envvar-default)
python/mxnet/model.py:192:0: R1711: Useless return at end of function or method (useless-return)
************* Module mxnet.gluon.rnn.rnn_cell
python/mxnet/gluon/rnn/rnn_cell.py:253:4: R0911: Too many return statements (7/6) (too-many-return-statements)
************* Module mxnet.image.detection
python/mxnet/image/detection.py:311:16: R0916: Too many boolean expressions in if statement (6/5) (too-many-boolean-expressions)
```

@vandanavk 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by vandanavk** [comment]
@marcoabreu

File: python/mxnet/ndarray/register.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 from ..base import mx_uint, check_call, _LIB, py_str, _init_op_module, _Null # pylint: disable=unused-import
 
 
+# pylint: disable=too-many-locals
```
Comment (marcoabreu) [inline python/mxnet/ndarray/register.py]: Obsolete due to global disable

File: python/mxnet/symbol/symbol.py [hunk 2]
```diff
@@ -1285,6 +1285,7 @@ def _get_ndarray_inputs(arg_key, args, arg_names, allow_missing):
             raise TypeError('Only accept list of NDArrays or dict of str to NDArray')
         return c_array(NDArrayHandle, arg_handles), arg_arrays
 
+    # pylint: disable=too-many-locals
```
Comment (marcoabreu) [inline python/mxnet/symbol/symbol.py]: Obsolete due to global disable

**Comment by cclauss** [comment]
Closing in favor of #12322

**Comment by cclauss** [comment]
Closing in favor of #12322

File: python/mxnet/ndarray/register.py (hunk 1 above)
Comment (cclauss) [reply python/mxnet/ndarray/register.py]: It defies explanation but these are actually required or the tests fail.

File: python/mxnet/symbol/symbol.py (hunk 2 above)
Comment (cclauss) [reply python/mxnet/symbol/symbol.py]: It defies explanation but these are actually required or the tests fail.

**Pull request closed without merging**